### PR TITLE
Require minimum versions on CMakeLists template.

### PR DESCRIPTION
**Pull request by MiguelCompany**
The code generated is now designed for Fast CDR >= 2.0 and Fast DDS >= 2.12, so the generated CMakeLists.txt should require those versions.

### CODE REVIEW

File: thirdparty/idl-parser
Comment (JLBuenoLopez-eProsima) [inline thirdparty/idl-parser:1]: I think the submodule has been mistakenly updated. Can you revert this change?

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Review by richiware — CHANGES_REQUESTED** [review]
Please, increase cmake_minimum_required to 3.22 as fastcdr and fastdds.
```
cmake_minimum_required(VERSION 3.22)
```

File: thirdparty/idl-parser
Comment (MiguelCompany) [reply thirdparty/idl-parser:1]: done

**Comment by MiguelCompany** [comment]
> Please, increase cmake_minimum_required to 3.22 as fastcdr and fastdds.
> 
> ```
> cmake_minimum_required(VERSION 3.22)
> ```

404ab20

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM
